adobe/react-spectrum
# fix(#2761): FocusScope: with no tabbable element in scope, restore focus to first focusable element

Fallback to include focusable elements when identifying the first focusable element in scope, if not tabbable elements are available in scope.

This is relevant for components like a Dialog, where when the contents of the FocusScope changes, and focus gets lost, we want to restore focus to the Dialog itself, which is focusable but not tabbable.


Closes #2761 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #2761.

## 📝 Test Instructions:

Test is meant to simulate a dialog whose contents change completely when an element within the dialog is clicked.

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/da566aa6219c6f6a8cab0123b860b36957b869d0/storybook/index.html?path=/story/focusscope--focusable-first-in-scope
2. Move keyboard focus to "Go to Dialog 2" button and use <kbd>Space</kbd> to click it.
3. Content should update and focus should go to the first tabbable element within the scope, the "Go to Dialog 3" button.
4. Use <kbd>Space</kbd> to click the "Go to Dialog 3" button.
5. Dialog 3 has no tabbable elements in scope, so focus will be set to the first focusable element in scope, the dialog itself.

## 🧢 Your Project:

Adobe/Accessibility
